runtime/CSharp/src/Recognizer.cs

```diff
@@ -264,7 +264,7 @@ public virtual string GetTokenErrorDisplay(IToken t)
         /// </exception>
         public virtual void AddErrorListener(IAntlrErrorListener<Symbol> listener)
         {
-            Args.NotNull("listener", listener);
+            if (listener == null) throw new NullReferenceException(nameof(listener));
```

Comment:
No! this changes the behavior (Args.NotNull throws a different exception and is shorter anyway)

Reply:
Thanks, it should be `ArgumentNullException`. `Args.NotNull` is used only one time, there is no need to keep entire class with method only for one line. Also, it's commonly used pattern in .NET, see https://referencesource.microsoft.com/#mscorlib/system/string.cs,86 for example.

Reply:
There is also no need to change working code if it does not improve it... it's a recipe for this kind of pitfall. Our test coverage is pretty poor, hence why both @parrt and myself are reluctant to big changes

Reply:
Anyway, Go and C++ runtimes were massively rewritten in the latest year without tests. And I doubt they strictly imitate Java code. It even might be not efficient.

Reply:
Go and C++ runtimes were rewritten to fix genuine problems. And bad practice from others is certainly no excuse for bad practice from oneself. Improvements in code coverage would be much more useful than improvements in code style.